### PR TITLE
Use reactphp/async instead of clue/reactphp-block

### PR DESCRIPTION
**Pull request by dinooo13**
See https://github.com/reactphp/async

Ref: https://github.com/clue/reactphp-block/pull/67, https://github.com/clue/reactphp-block/issues/68, https://github.com/reactphp/http/pull/464 and https://github.com/clue/reactphp-mq/pull/34

### CODE REVIEW

**Comment by dinooo13** [comment]
Still need to test this, no mysql:5 on arm.

**Comment by dinooo13** [comment]
Locally with PHP 8.1 this works fine for me. Can we restart the checks? Some weird failing there with 8.1, 8, 7.3 and HHVM green and only 7.1 failing.

File: tests/BaseTestCase.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ protected function createConnection(LoopInterface $loop)
         $factory = new Factory($loop);
         $promise = $factory->createConnection($this->getConnectionString());
 
-        return \Clue\React\Block\await($promise, $loop, 10.0);
+        return \React\Async\await($promise);
```
Comment (SimonFrings) [inline tests/BaseTestCase.php]: I'd suggest to keep the timeout like this:
```php
return \React\Async\await(\React\Promise\Timer\timeout($promise, 10.0));
```

**Review by SimonFrings — COMMENTED** [review]
@dinooo13 thanks for bringing this up :+1: 

One little thing below, can you also add some references in your PR description similar to my comment in [clue/reactphp-mq#34](https://github.com/clue/reactphp-mq/pull/34).

**Comment by dinooo13** [comment]
Resolved, thanks for the review 👍🏼

**Comment by dinooo13** [comment]
I think something is a bit flaky in the tests on the ci, should we maybe open an issue further investigating this?

**Review by SimonFrings — APPROVED** [review]
> I think something is a bit flaky in the tests on the ci, should we maybe open an issue further investigating this?

@clue restarted the failed tests, seems like there's been a hickup in one of them. From my end this PR seems fine as it is right now :+1:

**Review by clue — APPROVED** [review]
@dinooo13 Thank you for looking into this, changes LGTM! :+1: Happy to get this shipped with the next major as per #164, keep it up!

**Review by WyriHaximus — APPROVED** [review]

**Pull request merged**